### PR TITLE
Request for Code Review - Weather App Project (Using vanilla JavaScript and Weather API)

### PR DESCRIPTION
**Pull request by rloterh**
- Data is being fetched from the weather API correctly.
- Displays weather info: Temperature(Celcius/ Fahrenheit), Humidity and Wind speed
- Test cases are passing correctly

### CODE REVIEW

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Changes Requested :recycle:

## Hello Robert Loterh

THIS IS VERY IMPRESSIVE WORK. YOU HAVE DONE A GOOD JOB :+1:

## To Highlight :tada:

1. Your PR has a good title and description.
2. You have descriptive git commit messages.
3. Data is being fetched from the weather API correctly


However, there are some changes that you need to make in order to get your project approved

- [ ] Make development branch the default branch

- [ ] Kindly add a toggle control to determine if the result will be shown in Celsius or Fahrenheit [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png)

- [ ] Your app should change its appearance based on the received data [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png)

- [ ] It should have error handling for promises or async/await [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png)

[Error handling with promises](https://javascript.info/promise-error-handling)



<hr>

## OPTIONAL

- [ ] It would be nice if you add a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API

- [ ] It would be better if you fetch the current geolocation of the user when the app is first accessed and fetch the weather from that location


Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        this.apiKey}`,
+    )
+      .then((response) => {
+        if (!response.ok) {
+          alert('No weather found.');
+          throw new Error('No weather found.');
+        }
+        return response.json();
+      })
+      .then((data) => this.displayWeather(data));
```
Comment (MkrtichSargsyan) [inline src/index.js]: Add try-catch block for fetching data

**Comment by rloterh** [comment]
Implemented recommended changes required (Please note, I intend to implement the optional changes later):

- Made the development branch the default branch.

 - Added a toggle control to determine if the result will be shown in Celsius or Fahrenheit.

 - App changes its appearance based on the received data.

  - It has error handling for promises or async/await

**Comment by rloterh** [comment]
Implemented recommended changes required:
(Please note, I intend to implement the optional changes later, thanks)

- Made the development branch the default branch.

 - Added a toggle control to determine if the result will be shown in Celsius or Fahrenheit.

 - App changes its appearance based on the received data.

  - It has error handling for promises or async/await

**Review by oloomoses — APPROVED** [review]
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Hello @rloterh,
Great job working on the changes suggested by @MkrtichSargsyan.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

## Optional
- [ ] Before merging this pr. Kindly delete the commented code in the `index.js` file. Commented code makes your work look messy.

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

**Comment by rloterh** [comment]
Noted. Thanks @oloomoses

**Pull request merged**